x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/aggregate/TopListTests.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.aggregate;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.AbstractAggregationTestCase;
+import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class TopListTests extends AbstractAggregationTestCase {
+    public TopListTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseSupplier) {
+        this.testCase = testCaseSupplier.get();
+    }
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() {
+        var suppliers = List.of(
+            // All types
+            new TestCaseSupplier(List.of(DataType.INTEGER, DataType.INTEGER, DataType.KEYWORD), () -> {
+                var limit = randomIntBetween(2, 4);
+                return new TestCaseSupplier.TestCase(
+                    List.of(
+                        TestCaseSupplier.TypedData.multiRow(List.of(5, 8, -2, 0, 200), DataType.INTEGER, "field"),
```

Comment:
I strongly don't like the hard coded values here.  It's been a long and difficult refactor to move the scalar function tests over to using the data generation functions, and I don't want to go through that again.  The aggs tests should be using data generators from day one. (Data generators meaning stuff like `IntCases` and `DoubleCases` in `TestCaseSupplier`.)  We can make more of those if we need to, but I'd like one place where we're defining sets of values to test against, unless we have a very good reason only one aggregation needs to test against those values.

Reply:
I've created a branch with a proposal to use `<type>Cases()` with aggs. I'll open it after closing this one, as to avoid mixing more things